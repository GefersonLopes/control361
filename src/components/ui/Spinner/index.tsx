import clsx from "clsx";

import type { SpinnerProps } from "./types";

function Spinner({ className }: SpinnerProps) {
  return (
    <span
      className={clsx(
        "animate-spin rounded-full h-8 w-8",
        "border-4 border-gray-200 border-t-blue-500",
        className,
      )}
    ></span>
  );
}

export default Spinner;
