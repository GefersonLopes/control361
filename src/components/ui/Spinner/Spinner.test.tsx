import "@testing-library/jest-dom";

import { render } from "@testing-library/react";

import Spinner from "../Spinner";

describe("Spinner component", () => {
  it("renders a span element with default spinner classes", () => {
    const { container } = render(<Spinner />);
    const span = container.querySelector("span");
    expect(span).toBeInTheDocument();
    expect(span).toHaveClass(
      "animate-spin",
      "rounded-full",
      "h-8",
      "w-8",
      "border-4",
      "border-gray-200",
      "border-t-blue-500",
    );
  });

  it("renders an empty span without children", () => {
    const { container } = render(<Spinner />);
    const span = container.querySelector("span");
    expect(span).toBeEmptyDOMElement();
  });

  it("merges a custom className prop", () => {
    const { container } = render(
      <Spinner className="extra-class another-class" />,
    );
    const span = container.querySelector("span");
    expect(span).toHaveClass("extra-class", "another-class");
  });

  it("matches snapshot", () => {
    const { asFragment } = render(<Spinner className="snapshot-class" />);
    expect(asFragment()).toMatchSnapshot();
  });
});
