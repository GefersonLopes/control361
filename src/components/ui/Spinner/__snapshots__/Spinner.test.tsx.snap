// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner component matches snapshot 1`] = `
<DocumentFragment>
  <span
    class="animate-spin rounded-full h-8 w-8 border-4 border-gray-200 border-t-blue-500 snapshot-class"
  />
</DocumentFragment>
`;
